src/psyclone/domain/lfric/kernel/inter_grid_vector_arg_metadata.py

```diff
@@ -82,44 +79,32 @@ def create_from_fparser2(fparser2_tree):
             an InterGrid vector argument.
         :type fparser2_tree: :py:class:`fparser.two.Fortran2003.Part_Ref`
 
-        :returns: an instance of InterGridVectorArg.
+        :returns: an instance of InterGridVectorArgMetadata.
         :rtype: :py:class:`psyclone.domain.lfric.kernel.InterGridArg`
 
         '''
-        InterGridVectorArg.check_fparser2(
-            fparser2_tree, nargs=5, encoding=Fortran2003.Structure_Constructor)
-        InterGridArg.check_first_arg(
+        InterGridVectorArgMetadata.check_fparser2_arg(
```

Comment:
Looking at the duplication in the tests of this class and InterGridArgMetadata made me think that it shouldn't look like that. Could this be a class method and reuse most of the functionality from `super().xxxxx` to get everything except the `vector_length`? After all, this is just a vector of InterGridMetadata.

Reply:
I've refactored the meta_args argument classes and I think they are now much better. Hopefully you agree.